views.py

```diff
@@ -1081,6 +1086,210 @@ class RegistrationActiveView(HTTPMethodView):
+                return json({'error': 'I see what you did there. Not gonna happen.'}, 401)
+
+            # TODO: move valid_types to the LessonFeedbackQuestion class as a constant
+            valid_types = {
+                'open', 'abcd_single', 'abcd_multiple', 'int'
+            }
+            if data['type'] not in valid_types:
+                return json({'error': 'Invalid question type'}, 400)
+
+            if data['type'] in ['abcd_single', 'abcd_multiple'] and type(jloads(data['answers'])) != list:
+                return json({'error': 'Invalid answers provided (should be a list)'}, 400)
+
+            await question.update_from_dict(data)
+            return json({'msg': 'Question updated successfully'}, 200)
+        except DoesNotExist:
+            return json({'error': 'There is no question with given id'}, 404)
+
+    @user_required('admin')
+    async def delete(self, _, current_user, qid):
+        try:
+            await LessonFeedbackQuestion.get_first("id", qid)
+            await LessonFeedbackQuestion.detele_by_id(qid)
+
+            return json({}, 204)
+        except DoesNotExist:
+            return json({'error': 'There is no question with given id'}, 404)
+
+
+class LessonFeedbackMetaView(HTTPMethodView):
+    @user_required()
+    async def get(self, _, current_user, lid):
+        try:
+            await Lesson.get_first("id", lid)
+            questions = await LessonFeedbackQuestion.get_by_lesson_id(lid)
+
+            return json(questions)
+        except DoesNotExist:
+            return json({'error': 'A lesson with given id does not exist'}, 404)
+
+    @user_required('admin')
+    async def post(self, _, current_user, qid, lid):
+        try:
+            await LessonFeedbackQuestion.get_first("id", qid)
+        except DoesNotExist:
+            return json({'error': 'A question with given id does not exist'}, 400)
+
+        try:
+            await Lesson.get_first("id", lid)
+        except DoesNotExist:
+            return json({'error': 'A lesson with given id does not exist'}, 400)
+
+        meta = LessonFeedbackMeta(question=qid, lesson=lid)
+        await meta.create()
+
+        return json({'msg': 'Feedback association created successfully'}, 200)
+
+    @user_required('admin')
+    async def delete(self, _, current_user, qid, lid):
+        try:
+            # doing this twice since delete() could throw a weird out of bounds exc.
+            await LessonFeedbackMeta.get_first_by_many_field_value(
+                question=qid,
+                lesson=lid
+            )
+            await LessonFeedbackMeta.delete_by_many_fields(
+                question=qid,
+                lesson=lid
+            )
+
+            return json({}, 204)
+        except DoesNotExist:
+            return json({'error': 'No association with given constraints exists in the database'}, 404)
+
+
+class LessonFeedbackAnswerView(HTTPMethodView):
+    async def _validate_answers(self, data):
```

Comment:
this could be static

Reply:
or event moved to model part ;)

Reply:
I'd go with static as this is more view-related than model-related IMHO. I'll fix that.